wordpress-mobile/WordPress-Android
# Remove Blogging reminders feature config

This PR 
- Updates Blogging Reminder feature to shown only on Jetpack 
- removes the `BloggingReminderFeatureConfig`, since it has been released over an year ago.
- Updates Site Settings to reflect the change

The Blogging Reminders feature has two main UI components:

A one-time prompt shown per-blog to promote the feature.
A setting in Site Settings for the user to adjust when the local reminder notification is shown.

To test:

The Blogging Reminders feature has two main UI components:

A one-time prompt shown per-blog to promote the feature.
A setting in Site Settings for the user to adjust when the local reminder notification is shown.

JP app
- Add a new site to the Jetpack app
- Publish a new Post (FAB -> Blog post)
- Confirm that the Blogging Reminders prompt is shown
- Additionally, navigate to Menu -> Site Settings -> Verify Reminders setting is visible under **Blogging** heading, and works

WP app
- Add a site to WP app
- Publish a Post (FAB -> Blog post
- Confirm that the Blogging Reminders prompt is not shown
- Additionally, navigate to Menu -> Site Settings -> Verify no Reminders setting is shown


## Regression Notes
1. Potential unintended areas of impact
Tested on both WP and JP apps

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing, and unit tests

3. What automated tests I added (or what prevented me from doing so)
Updated unit tests

PR submission checklist:
